File: C++/SelectionSort.cpp
#include<iostream>
using namespace std;

void SelectionSort(int arr[],int n){
    if (n<=1) return;
    int smallestindex=0, i;
    for (i=1; i<n;i++){
        if (arr[i]<arr[smallestindex]) smallestindex=i;
    }
    swap(arr[0],arr[smallestindex]);
    SelectionSort(arr+1,n-1);
}
int main(){
 int n;
 cout<<"Enter size of the array"<<endl;
 cin>>n;
 int arr[n];
 cout<<"Enter Elements of the array"<<endl;
 for (int i=0;i<n;i++) cin>>arr[i];
 SelectionSort(arr,n);
 cout<<"Sorted array using SelectionSort algorithm is"<<endl;
 for (int i=0;i<n;i++) cout<<arr[i]<<" ";
 cout<<endl;
}

File: Java/Selectionsort.java
import java.util.Scanner;
class co
{
    static void selection_sort(int m[])
    {   int a=m.length;
        for(int i=0;i<a-1;i++)
        {  int min=i;
            for(int j=i+1;j<a;j++)
            { if(m[j]<m[min])
                min=j;
            }
            int aa=m[min];
            m[min]=m[i];
            m[i]=aa;
        }
    }
    public static void main(String args[])
    {
        Scanner sc=new Scanner (System.in);
        System.out.println("enter no. of elements in an array");
        int a=sc.nextInt();
        int arr[]=new int [a];
        for (int i=0;i<a;i++)
        arr[i]=sc.nextInt();
        selection_sort(arr);
        for(int mi:arr)
         System.out.print(mi+" ");
    }}
    
        